commit fa16a9099b95618d56efd3ca151a9c05d1e99e7e
Author: Matthias Dellweg <mdellweg@redhat.com>
Date:   Wed May 5 15:27:10 2021 +0200

    Fixed "connection already closed" error in the Registry handler.
    
    backports #8672
    
    fixes #8697
    
    (cherry picked from commit 756432fac6e0b188ff80b715ae33509f25103d48)

diff --git a/CHANGES/8697.bugfix b/CHANGES/8697.bugfix
new file mode 100644
index 0000000..962b824
--- /dev/null
+++ b/CHANGES/8697.bugfix
@@ -0,0 +1,2 @@
+Fixed "connection already closed" error in the Registry handler.
+(backported from #8672)
diff --git a/pulp_container/app/registry.py b/pulp_container/app/registry.py
index 0849a87..ac9a200 100644
--- a/pulp_container/app/registry.py
+++ b/pulp_container/app/registry.py
@@ -97,6 +97,8 @@ class Registry(Handler):
                 streamed back to the client.
 
         """
+        self._reset_db_connection()
+
         path = request.match_info["path"]
         tag_name = request.match_info["tag_name"]
         distribution = self._match_distribution(path)
@@ -203,6 +205,7 @@ class Registry(Handler):
         """
         Return a response to the "GET" action.
         """
+        self._reset_db_connection()
 
         path = request.match_info["path"]
         digest = "sha256:{digest}".format(digest=request.match_info["digest"])

